django/db/backends/postgresql/base.py

```diff
@@ -297,6 +298,18 @@ def init_connection_state(self):
             if not self.get_autocommit():
                 self.connection.commit()
 
+        # Set the role on the connection. This is useful if the
+        # credential used to login is not the same as the role that
+        # owns database resources. As can be the case when using
+        # temporary or ephemeral credentials.
+        new_role = self.settings_dict["OPTIONS"].get("assume_role")
+        if new_role:
+            with self.connection.cursor() as cursor:
+                cursor.execute("SET ROLE %s", (new_role,))
+
+            if not self.get_autocommit():
```

Comment:
Given that this duplicates the code from above and would result in two commits (in case of timezone changes as well), I am wondering if it would make more sense to put this into a variable (`should_commit=True`) and then after those two blocks have a:
```py
if should_commit and not self.get_autocommit():
  self.connection.commit()
```

Reply:
I opted for a slightly bigger refactor to make this more in-line with `ensure_timezone` but have also eliminated the multiple-commits issue.